fog/sql_recovery_db/src/lib.rs

```diff
@@ -1005,21 +1003,21 @@ impl SqlRecoveryDb {
             use schema::ingested_blocks::dsl;
             dsl::ingested_blocks
                 .filter(dsl::ingress_public_key.eq(key_bytes))
-                .filter(dsl::block_number.ge(block_index as i64))
-                .limit(block_count as i64)
+                .filter(dsl::block_number.ge(block_range.start_block as i64))
+                .limit(block_range.len() as i64)
                 .select((dsl::block_number, dsl::proto_ingested_block_data))
                 .order(dsl::block_number.asc())
         };
 
         // We will get one row for each hit in the table we found
         let rows: Vec<(i64, Vec<u8>)> = query.load(&conn)?;
 
-        if rows.len() > block_count {
+        if rows.len() > (block_range.len() as usize) {
```

Comment:
:upside_down_face: it's already a usize
```suggestion
        if rows.len() > block_range.len() {
```